lib/CodeGen/CGDynamicCheck.cpp

```diff
@@ -111,6 +111,66 @@ void CodeGenFunction::EmitDynamicBoundsCheck(const Address PtrAddr, const Bounds
   EmitDynamicCheckBlocks(Builder.CreateAnd(LowerChk, UpperChk, "_Dynamic_check.range"));
 }
 
+void CodeGenFunction::EmitDynamicBoundsCheck(const BoundsExpr *DynamicBounds,
```

Comment:
Could you rename `DynamicBounds` to `CastBounds` and `SrcBounds` to `SubExprBounds`?   That will make it easier to remember which bounds are which?